samples/25.message-reaction/src/main/java/com/microsoft/bot/sample/messagereaction/ActivityLog.java

```diff
@@ -0,0 +1,41 @@
+package com.microsoft.bot.sample.messagereaction;
+
+import com.microsoft.bot.builder.Storage;
+import com.microsoft.bot.schema.Activity;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+public class ActivityLog {
+
+    private Storage _storage;
+
+    public void ActivityLog(Storage storage) {
+        _storage = storage;
+    }
+
+    public CompletableFuture<Void> Append(String activityId, Activity activity) throws IllegalAccessException {
+        if (activityId == null) {
+            throw new IllegalAccessException("activityId");
```

Comment:
Aren't there some better exceptions for missing argument in Java?

Reply:
Well I think you are right, maybe NullPointerException would be more appropiate

Reply:
looking at the other samples, IllegalArgumentException seems to be the choice